examples/custom_protocol.rs

```diff
@@ -28,36 +29,18 @@ fn main() -> wry::Result<()> {
 
   let _webview = WebViewBuilder::new(window)
     .unwrap()
-    .with_custom_protocol("wry".into(), move |request| {
-      let path = request.uri().path();
-      // Read the file content from file path
-      let content = if path == "/" {
-        PAGE1_HTML.into()
-      } else {
-        // `1..` for removing leading slash
-        read(canonicalize(PathBuf::from("examples").join(&path[1..]))?)?.into()
-      };
-
-      // Return asset contents and mime types based on file extentions
-      // If you don't want to do this manually, there are some crates for you.
-      // Such as `infer` and `mime_guess`.
-      let mimetype = if path.ends_with(".html") || path == "/" {
-        "text/html"
-      } else if path.ends_with(".js") {
-        "text/javascript"
-      } else if path.ends_with(".png") {
-        "image/png"
-      } else if path.ends_with(".wasm") {
-        "application/wasm"
-      } else {
-        unimplemented!();
-      };
-
-      Response::builder()
-        .header(CONTENT_TYPE, mimetype)
-        .body(content)
-        .map_err(Into::into)
-    })
+    .with_asynchronous_custom_protocol(
+      "wry".into(),
+      move |request, response| match get_wry_response(request) {
```

Comment:
Could we also change the examples and documentation to show that the second parameter is a responder and not an actual response ?
```suggestion
      move |request, responder| match get_wry_response(request) {
```